File: backend/internal/models/post_comment.go
package models

import "time"

// PostComment is one row in post_comments. Flat — no parent_id reference,
// since v1 doesn't support reply threads.
type PostComment struct {
	ID         int64      `json:"id"`
	PostID     int64      `json:"postId"`
	AuthorID   int64      `json:"authorId"`
	AuthorName string     `json:"authorName"`
	Body       string     `json:"body"`
	CreatedAt  time.Time  `json:"createdAt"`
	// EditedAt is non-nil when the author has changed the body since
	// posting. The UI renders "(edited)" next to the timestamp.
	EditedAt   *time.Time `json:"editedAt,omitempty"`
}

File: lib/models/post_comment.dart
/// One row in the comments thread of a post. Mirrors the Go
/// `models.PostComment` struct.
class PostComment {
  final int id;
  final int postId;
  final int authorId;
  final String authorName;
  final String body;
  final DateTime createdAt;

  /// Non-null when the author has edited the body since posting (mig 0037).
  /// Render "(edited)" next to the timestamp in the UI to be transparent
  /// about modifications.
  final DateTime? editedAt;

  const PostComment({
    required this.id,
    required this.postId,
    required this.authorId,
    required this.authorName,
    required this.body,
    required this.createdAt,
    this.editedAt,
  });

  bool get wasEdited => editedAt != null;

  factory PostComment.fromJson(Map<String, dynamic> json) => PostComment(
        id: (json['id'] as num).toInt(),
        postId: (json['postId'] as num?)?.toInt() ?? 0,
        authorId: (json['authorId'] as num?)?.toInt() ?? 0,
        authorName: json['authorName'] as String? ?? '',
        body: json['body'] as String? ?? '',
        createdAt: DateTime.tryParse(json['createdAt'] as String? ?? '') ??
            DateTime.now(),
        editedAt: json['editedAt'] != null
            ? DateTime.tryParse(json['editedAt'] as String)
            : null,
      );
}
